fhirclient/auth.py

```diff
@@ -294,9 +294,8 @@ def _request_access_token(self, server, params):
         # The refresh token issued by the authorization server. If present, the
         # app should discard any previous refresh_token associated with this
         # launch, replacing it with this new value.
-        refresh_token = ret_params.get('refresh_token')
-        if refresh_token is not None:
-            self.refresh_token = refresh_token
+        self.refresh_token = ret_params.get('refresh_token') or params.get('refresh_token')
```

Comment:
Can you have instead the following?

```
refresh_token = ret_params.get('refresh_token') or params.get('refresh_token')
if refresh_token is not None:
    self.refresh_token = refresh_token
```

Reply:
Since the `or` operator returns None for `None or None` they would accomplish the same thing but with extra unneeded code that decreases the readability:
```python
Python 3.7.6 (default, Feb 13 2020, 15:54:11) 
[Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> ret_params = {}; params = {};
>>> refresh_token = ret_params.get('refresh_token') or params.get('refresh_token')
>>> print(refresh_token)
None
>>> 
```

Reply:
The thing is self.refresh_token was being set only if not None, otherwise was left with existing value.
But your change could set it to None.

Reply:
Yes true. 

hmm..    I found in practice (downloads from the Cerner, Epic and HAPI servers and many health systems) that the submitted line of code works.  I don't think there could be a scenario where some system doesn't provide the refresh_token if it expects you to use the prior one, if a system is supporting refresh tokens it is providing them with every call.

The prior code was incorrect and not tested in real world scenarios, so the goal of maintaining requests with an old refresh token is dubious.  Failing fast would be preferred.   I know it was difficult to debug (at least 2 hours for me to notice) because it took a lot of stepping through to notice that the refresh token the code was attempting to use was from a previous call and not the current one.    It's really hard to spot that a token string was replaced in the incoming request but not assigned, because they are two seemingly random sequences if you are not looking closely (another argument for allowing the None assignment).

I don't have a reference to cite at the moment but I think if a system is supporting refresh tokens it must always return the request token with every call.

Reply:
I suppose there could be servers I've not tested that only return the Refresh token with the first authentication.  I think my app would continue to work with either of the code suggestions in place.  If you feel we should use your suggested code, let me know and I'll push a change after some local testing.

Reply:
Please do a local test for this, it should cover both your scenario and the way it's working now.

```
refresh_token = ret_params.get('refresh_token') or params.get('refresh_token')
if refresh_token is not None:
    self.refresh_token = refresh_token
```

Reply:
latest commit addresses these concerns